test/Microsoft.AspNetCore.Mvc.TagHelpers.Test/InputTagHelperTest.cs

```diff
@@ -457,6 +457,77 @@ public void Process_GeneratesFormattedOutput(string specifiedType, string expect
             Assert.Equal(expectedTagName, output.TagName);
         }
 
+        [Theory]
+        [InlineData("datetime", "datetime")]
+        [InlineData(null, "datetime-local")]
```

Comment:
FYI this demonstrates the default is still `datetime-local` for `DateTime` properties.